File: src/main/java/com/zoarial/iot/action/model/IoTActionArgumentList.java
package com.zoarial.iot.action.model;

import java.util.ArrayList;

public class IoTActionArgumentList extends ArrayList<IoTActionArgument> {


    public IoTActionArgumentList() {
        super();
    }
    public IoTActionArgumentList(int len) {
        super(len);
    }

    public boolean add(String str) {
        return super.add(new IoTActionArgument(str));
    }

    public boolean add(int i) {
        return super.add(new IoTActionArgument(i));
    }

    public boolean add(long l) {
        return super.add(new IoTActionArgument(l));
    }

    public boolean add(boolean b) {
        return super.add(new IoTActionArgument(b));
    }



}

File: src/main/java/com/zoarial/iot/action/model/IoTActionArgumentList.kt
package com.zoarial.iot.action.model

class IoTActionArgumentList : ArrayList<IoTActionArgument> {
    constructor() : super() {}
    constructor(len: Int) : super(len) {}

    fun add(str: String): Boolean {
        return super.add(IoTActionArgument(str))
    }

    fun add(i: Int): Boolean {
        return super.add(IoTActionArgument(i))
    }

    fun add(l: Long): Boolean {
        return super.add(IoTActionArgument(l))
    }

    fun add(b: Boolean): Boolean {
        return super.add(IoTActionArgument(b))
    }
}